src/tracker/mod.rs

```diff
@@ -1,6 +1,78 @@
+pub mod core;
 pub mod key;
 pub mod mode;
 pub mod peer;
 pub mod statistics;
 pub mod torrent;
-pub mod tracker;
+
+pub mod helpers {
+    use std::net::IpAddr;
+    use std::sync::Arc;
+
+    use super::mode::TrackerMode;
+    use crate::databases::database::{self, Database};
+    use crate::settings::{
+        CommonSettings, CommonSettingsBuilder, DatabaseSettingsBuilder, GlobalSettings, GlobalSettingsBuilder, LogFilterLevel,
+    };
+
+    pub struct TrackerArgs {
```

Comment:
@da2ce7 I think this struct is useful now, but in the long term, I think we should extract some responsibilities from the Tracker, for example persistence. I think we could have:

- One handler or controller fer each request type (announce, scrape, ...)
- An app service (announce is the same from HTTP and udp). It could be also a command or query with its handler. For example: AnnounceCommand with AnnounceCommandHandler.
- A TorrentService to add/update/remove torrents. We can trigger events at this level.
- The TorrentService can use a TorrentRepository (DB: SQLite or MySQL) and a TorrentCache (the current in-memory structure).
But it's just a vague idea I have. I wanted to draft something, but I have not had time. And I also wanted to get to know the repo better before spending time proposing this kind of change.